MaterialDesignThemes.Wpf/MaterialDateDisplay.cs

```diff
@@ -14,11 +14,19 @@ static MaterialDateDisplay()
 
         public MaterialDateDisplay()
         {
-            SetCurrentValue(DisplayDateProperty, DateTime.Now.Date);
+            SetCurrentValue(DisplayDateProperty, DateTime.Today);
+
+            UpdateComponents();
         }
 
         public static readonly DependencyProperty DisplayDateProperty = DependencyProperty.Register(
-            nameof(DisplayDate), typeof (DateTime), typeof (MaterialDateDisplay), new PropertyMetadata(default(DateTime), DisplayDatePropertyChangedCallback));
+            nameof(DisplayDate),
+            typeof (DateTime),
+            typeof (MaterialDateDisplay),
+            new PropertyMetadata(
+                System.Threading.Thread.CurrentThread.CurrentUICulture.Calendar != null ? System.Threading.Thread.CurrentThread.CurrentUICulture.Calendar.MinSupportedDateTime : default(DateTime),
```

Comment:
I don't think we want to have different default value based on culture. Rather I think we should take this culture into account at the places where the `DisplayDate` property is used.

Reply:
Yea I think that would be better. Just controlling it with a default value, also doesn't guard against someone setting a `DateTime` that is outside of the range. 
I think in these cases I am happy simply adjusting the invalid date to be in the valid range. I would also favor using the Language setting from the control over the culture specified on the thread. Something like:
```C#
MaterialDateDisplay materialDate = ...;
CultureInfo culture = materialDate.Language.GetSpecificCulture();
var minDateTime = culture.Calendar.MinSupportedDateTime;
var maxDateTime = culture.Calendar.MaxSupportedDateTime;

DateTime value = (DateTime) basevalue;
if (value < minDateTime)
{
    return minDateTime;
}

if (value > maxDateTime)
{
    return maxDateTime;
}

return value;
```

Reply:
I implemented your suggestions.